tf2_ros/CMakeLists.txt

```diff
@@ -1,97 +1,156 @@
   src/static_transform_broadcaster.cpp
 )
-add_dependencies(${PROJECT_NAME} tf2_msgs_gencpp)
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES})
 
-# buffer_server executable
-add_executable(${PROJECT_NAME}_buffer_server src/buffer_server_main.cpp)
-add_dependencies(${PROJECT_NAME}_buffer_server tf2_msgs_gencpp)
-target_link_libraries(${PROJECT_NAME}_buffer_server
-  ${PROJECT_NAME}
-  ${catkin_LIBRARIES}
-  ${Boost_LIBRARIES}
-)
-set_target_properties(${PROJECT_NAME}_buffer_server
-  PROPERTIES OUTPUT_NAME buffer_server
-)
+target_link_libraries(${PROJECT_NAME}
+  ${geometry_msgs_LIBRARIES}
+  ${rclcpp_LIBRARIES}
+  ${rmw_LIBRARIES}
+  ${rmw_implementation_LIBRARIES}
+  ${tf2_LIBRARIES}
+  ${tf2_msgs_LIBRARIES}
+  )
+# # buffer_server executable
+# add_executable(buffer_server src/buffer_server_main.cpp)
+# add_dependencies(${PROJECT_NAME}_buffer_server tf2_msgs_gencpp)
+# target_link_libraries(${PROJECT_NAME}_buffer_server
+#   ${PROJECT_NAME}
+#   ${catkin_LIBRARIES}
+#   ${Boost_LIBRARIES}
+# )
+
 
 # static_transform_publisher
-add_executable(${PROJECT_NAME}_static_transform_publisher
+add_executable(static_transform_publisher
   src/static_transform_broadcaster_program.cpp
 )
-add_dependencies(${PROJECT_NAME}_static_transform_publisher tf2_msgs_gencpp)
-target_link_libraries(${PROJECT_NAME}_static_transform_publisher
+target_link_libraries(static_transform_publisher
   ${PROJECT_NAME}
-  ${catkin_LIBRARIES}
-)
-set_target_properties(${PROJECT_NAME}_static_transform_publisher
-  PROPERTIES OUTPUT_NAME static_transform_publisher
+  ${geometry_msgs_LIBRARIES}
+  ${rclcpp_LIBRARIES}
+  ${rmw_LIBRARIES}
+  ${rmw_implementation_LIBRARIES}
+  ${tf2_LIBRARIES}
+  ${tf2_msgs_LIBRARIES}
+  )
+
+
+add_executable(tf2_echo
+  src/tf2_echo.cpp
 )
+target_link_libraries(tf2_echo
+  ${PROJECT_NAME}
+  ${geometry_msgs_LIBRARIES}
+  ${rclcpp_LIBRARIES}
+  ${rmw_LIBRARIES}
+  ${rmw_implementation_LIBRARIES}
+  ${tf2_LIBRARIES}
+  ${tf2_msgs_LIBRARIES}
+  )
+
+  add_executable(tf2_monitor
+    src/tf2_monitor.cpp
+  )
+  target_link_libraries(tf2_monitor
+    ${PROJECT_NAME}
+    ${geometry_msgs_LIBRARIES}
+    ${rclcpp_LIBRARIES}
+    ${rmw_LIBRARIES}
+    ${rmw_implementation_LIBRARIES}
+    ${tf2_LIBRARIES}
+    ${tf2_msgs_LIBRARIES}
+    )
 
 # Install rules
 install(TARGETS
-  ${PROJECT_NAME} ${PROJECT_NAME}_buffer_server
-  ${PROJECT_NAME}_static_transform_publisher
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+  ${PROJECT_NAME}
+  tf2_echo
+  tf2_monitor
+  # buffer_server
+  static_transform_publisher
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION lib/${PROJECT_NAME}
 )
 
 install(DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+  DESTINATION include/${PROJECT_NAME}
 )
 
 
 # Tests
-if(CATKIN_ENABLE_TESTING)
-
+# if(AMENT_ENABLE_TESTING)
+if (false)
```

Comment:
TODO added. They are close to being able to be ported. They use some things that are not available yet such as sim_time.